core/lib/server/pub.ts

```diff
@@ -1354,6 +1374,68 @@ export async function getPubsWithRelatedValuesAndChildren<
 									)
 								)
 								.leftJoin("PubsInStages", "pubs.id", "PubsInStages.pubId")
+								.where((eb) =>
+									eb.exists(
+										eb.selectFrom("capabilities" as any).where((ebb) => {
+											type Inner =
+												typeof ebb extends ExpressionBuilder<
+													infer Thing,
+													any
+												>
+													? Thing
+													: never;
+											const eb = ebb as ExpressionBuilder<
+												Inner & {
+													capabilities: {
+														membId: string;
+														type: MembershipType;
+														role: MembershipCapabilitiesRole;
+													};
+												},
+												any
+											>;
+
+											return eb.or([
```

Comment:
this big condition is repeated here in the recursive query, and initially in the root pubs selection. and what it does is just check if the user has the capability viewPub on each pub or viewStage on that pub's stage. we need this big set of checks to make sure that only the capabilities that are relevant for any given pub are used, since the "capabilities" cte has all of the user's pub and stage capabilities, not filtered by any pub or stage.